src/lex/tests.rs

```diff
@@ -46,7 +46,28 @@ where
 }
 
 fn match_char(lexed: Option<LexType>, expected: u8) -> bool {
-    match_data(lexed, |c| c == Ok(&LiteralToken::Char(expected).into()))
+    use crate::data::hir::LiteralValue;
+    match lexed {
+        Some(Ok(loc)) => {
+            if let Locatable {
+                data: Token::Literal(LiteralToken::Char(rcstr)),
+                location,
+            } = loc
+            {
+                let loc = Locatable {
+                    data: LiteralToken::Char(rcstr),
+                    location,
+                };
+                match loc.parse().data {
+                    LiteralValue::Char(c) => c == expected,
+                    _ => false,
+                }
+            } else {
+                return false;
+            }
+        }
+        _ => false,
+    }
```

Comment:
Implementing `.parse()` on the `Locatable` is a little odd, can you implement it on `LiteralToken` instead?

Also, why do you match on the value after parsing it? Shouldn't it always be a `Char`? It might be cleaner to implement a standalone parsing function that doesn't go through `LiteralValue`.

```suggestion
    match lexed {
        Some(Ok(Locatable {
            data: Token::Literal(literal @ LiteralToken::Char(_)),
            location,
        } =>  match literal.parse() {
                    LiteralValue::Char(c) => c == expected,
                    _ => false,
                }
        _ => false,
    }
```